plenoptic/simulate/models/portilla_simoncelli.py

```diff
@@ -0,0 +1,1072 @@
+                pixel_statistics
+                + magnitude_means
+                + auto_correlation_magnitude
+                + skew_reconstructed
+                + kurtosis_reconstructed
+                + auto_correlation
+                + cross_orientation_correlation_magnitude
+                + cross_scale_correlation_magnitude
+                + cross_orientation_correlation_real
+                + cross_scale_correlation_real
+                + var_highpass_residual
+            )
+
+        return scales
+
+    def forward(self, image, scales=None):
+        r"""Generate Texture Statistics representation of an image (see reference [1])
+
+        Parameters
+        ----------
+        image : torch.tensor
+            A tensor containing the image to analyze. We want to operate
+            on this in the pytorch-y way, so we want it to be 4d (batch,
+            channel, height, width). If it has fewer than 4 dimensions,
+            we will unsqueeze it until its 4d
+
+        Returns
+        =======
+        representation_vector: torch.tensor
+            A flattened tensor (1d) containing the measured representation statistics.
+
+        """
+        device = image.device
+        while image.ndimension() < 4:
+            image = image.unsqueeze(0)
+
+        # self.pyr.to(image.dtype)
+        self.pyr_coeffs = self.pyr.forward(image)
+        self.representation = OrderedDict()
+
+        ### SECTION 1 (STATISTIC: pixel_statistics) ##################
+        #  Calculate pixel statistics (mean, variance, skew, kurtosis, min, max).
+        self.representation["pixel_statistics"] = OrderedDict()
+        self.representation["pixel_statistics"]["mean"] = torch.mean(image)
+        self.representation["pixel_statistics"]["var"] = torch.var(image)
+        self.representation["pixel_statistics"]["skew"] = PortillaSimoncelli.skew(
+            image
+        )
+        self.representation["pixel_statistics"][
+            "kurtosis"
+        ] = PortillaSimoncelli.kurtosis(image)
+        self.representation["pixel_statistics"]["min"] = torch.min(image)
+        self.representation["pixel_statistics"]["max"] = torch.max(image)
+
+        ### SECTION 2 (STATISTIC: mean_magnitude) ####################
+        # Calculate the mean of the magnitude of each band of pyramid
+        # coefficients.  Additionally, this section creates two
+        # other dictionaries of coefficients: magnitude_pyr_coeffs
+        # and real_pyr_coeffs, which contain the magnitude of the
+        # pyramid coefficients and the real part of the pyramid
+        # coefficients respectively.
+        self.representation["magnitude_means"] = self._calculate_magnitude_means()
+
+        ### SECTION 3 (STATISTICS: auto_correlation_magnitude,
+        #                          skew_reconstructed,
+        #                          kurtosis_reconstructed,
+        #                          auto_correlation_reconstructed) #####
+        #
```

Comment:
I've updated this comment to explain why they're all computed together (see below as well) and I'm not sure what you mean when you say that it's not right above the place where the stats are computed it.  It would break with the consistency of the commenting style to move them to the method calculate_autocorrelation_skew_kurtosis()  so I'm leaving them where they are above the intialization of the variables which is followed by the method that calculates those values.

Reply:
I also think that when I change PortillaSimoncelli so that it doesn't contain self.representation, the clarity around where the stats are calculated will be improved.  This will occur in a separate, subsequent pull request.